world-chain-builder/crates/world/payload/src/builder.rs

```diff
@@ -616,11 +606,20 @@ where
             }
 
             // If the transaction is verified, check if it can be added within the verified gas limit
-            if pooled_tx.valid_pbh()
-                && info.cumulative_gas_used + tx.gas_limit() > verified_gas_limit
-            {
-                best_txs.mark_invalid(tx.signer(), tx.nonce());
-                continue;
+            if let Some(payloads) = pooled_tx.pbh_payload() {
+                if info.cumulative_gas_used + tx.gas_limit() > verified_gas_limit {
+                    best_txs.mark_invalid(tx.signer(), tx.nonce());
+                    continue;
+                }
+
+                if payloads
+                    .iter()
+                    .any(|payload| spent_nullifier_hashes.contains(&payload.nullifier_hash))
```

Comment:
I think this may allow an edge case where a bundler provides `n` PBH UserOps all with the same PBH Payload/nullifier.